### ISSUE TITLE
Identity information

### ISSUE THREAD

**Issue by daniele-athome** [kontalk/androidclient#456]
Spin-off the personal key activity or use a dialog with identity information (uid, phone number, fingerprint)
##

**Comment by juanito003** [comment]
Also it would be nice to be able to copy/paste the fingerprint of my personal key, in case I want to verify it with someone abroad.

**Comment by breversa** [comment]
Also, is there any way to see/edit my user name after the initial registration ? It's not that critical, since my buddies can change my display name in their contact app, but what if I put a random name at setup and wish to clean it up ?

**Comment by daniele-athome** [comment]
You can see your user name in the "My personal key" activity, from the main screen menu.
I'm sorry you can't edit it yet. It's a bit complicated because it involves creating a new uid in your PGP key and it needs something on the server to help too.

**Comment by breversa** [comment]
Thanks for the info.
In that case, I guess a warning would be useful, such as "Warning : you won't be able to change your user name once you set it up, so be careful".

**Comment by lutzke** [comment]
The name "My Personal Key" sounds way too much like "private key," when this is clearly nothing of the sort (why would you share your private key with your friends, or anyone?) 

I'd like to see others suggest alternative names for "My Personal Key." I've come up with two so far, each with their merits and drawbacks:

1. Former BBM users (a rather large group of people) are familiar with "Barcodes," so we could call it "My Code." Drawback: The name "My Code" doesn't make it very clear that this is a security thing related to encryption.
2. Telegram calls it "Encryption Key." I think most people know what the word encryption means, so this shouldn't be Greek to most people. This one tells the user what they're actually seeing. It's technical, without being too technical. (Calling it a "public key" would probably confuse a lot of people.)

If it were possible to add a user with their public key (it's not, currently) then I think number one would be the better choice. Users would, I think, be much more willing to tell their friends, "here's my code," rather than, "here's my encryption key."

Given that in the current build, "My Personal Key" is a public key used for encryption, and nothing else, I am suggesting that it be renamed to "Encryption Key."

**Comment by abika** [comment]
TL;DR: using "Personal Key" is imo totally fine.

When choosing a word that is used to "translate" complex technical terms so that the user can "understand" it multiple aspects are important:
1. It should be easy to understand for the average user. In this case somebody who hasn't heard anything about asymmetric cryptography should have an idea what we are describing here. 
2. Those who have some previous knowledge should also know what is meant. In this case the simple fact that its a (cryptography) key.
3. It should be precise. Trying to invent some different analogy or using a only slightly related term makes it even harder to understand and will certainly clash with (2).

Now, because of (2) and (3) using something like "(security/my/bar-) code" is not an option. Cause it is a key, point.

And the fact that its actually a key pair with a public and a private key (actually a key pair ring consisting of multiple key pairs) and both are connected to each other, one is used for encryption and validation and the other one for decryption and signing... should be hidden for the user. Cause its not important and the average user doesn't care (see (1)). It is only important that it works.

What follows is that "Encryption Key" is not precise (see (3)). The dialog shows the fingerprint of the public key but it identifies the whole key. Using "encryption" here can also be misleading as this (=your) key is not used for encryption. You are using the public key of the receiver (that's why PGP folks use "public"/"private" and not "encryption"/"decryption": to prevent confusion).
And also note that "personal key" is used multiple times in the "maintenance" settings menu. And for all cases the whole key ring is meant. Using something else here would be wrong.

My conclusion is that @daniele-athome did already a great job at user friendly usability and we shouldn't change it.

**Comment by lutzke** [comment]
@abika You make a number of excellent points. I must say that after a bit of thought, I agree with your conclusion.

**Comment by daniele-athome** [comment]
I must say, finding a nice and easy name for the key pair was not easy. There are so many technical aspects involved (most of them perfectly explained by @abika) that it's hard to find a single word to make the user understand at a glance what it's about.
@lutzke I'm glad you understand, thanks for your contribution - ideas are always precious in a project like this.
